src/parser/smt2/smt2_state.cpp

```diff
@@ -142,6 +142,8 @@ void Smt2State::addFiniteFieldOperators()
 
 void Smt2State::addDatatypesOperators()
 {
+  TermManager& tm = d_solver->getTermManager();
```

Comment:
Ditto, use the cached term manager in this class (which inherits from ParserState).